### ISSUE TITLE
Parsing an anonymous function generates "mismatched input" error

### ISSUE THREAD

**Issue by Teloah** [fabriciocolombo/sonar-delphi#18]
Analyzing a unit with an anonymous function generates the following error:

```
19:31:09.270 INFO  - Parsing project Default Project
19:31:09.271 INFO  - Files to parse: 1
C:\Develop\tmp_work\sonar-test\src\AnonymousFuncTest.pas line 20:26 mismatched input '(' expecting END
19:31:09.640 INFO  - 0% done...
19:31:09.641 INFO  - 10% done...
```

``` delphi
unit AnonymousFuncTest;

interface

type
  TStringPredicate = reference to function(const aValue : string) : Boolean;

function StringMatches(const aValue : string; aPredicate : TStringPredicate) : Boolean;
function IsStringEmpty(const aValue : string) : Boolean;

implementation

function StringMatches(const aValue : string; aPredicate : TStringPredicate) : Boolean;
begin
  Result := aPredicate(aValue);
end;

function IsStringEmpty(const aValue : string) : Boolean;
begin
  Result := StringMatches(aValue,
    function(const aValue : string) : Boolean
    begin
      Result := aValue = '';
    end);
end;

end.
```

**Comment by fabriciocolombo** [comment]
I think this issue has been solved by commit 6a1bddf0b95e9433ed82964a593d7ee81e03fcb9.
I wrote a test to ensure it works.

**Comment by fabriciocolombo** [comment]
I published the version [0.3.3-SNAPSHOT](https://github.com/fabriciocolombo/sonar-delphi/releases/tag/0.3.3-SNAPSHOT). Can you try it, please?

**Comment by Teloah** [comment]
Yes, now anonymous functions are processed and some violations that I tried are properly generated inside them.
